src/movies.js

```diff
@@ -1,16 +1,150 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+
+function getAllDirectors(movies){
+    return movies.map((movie) => movie.director).filter((value, index, self) => self.indexOf(value) === index);
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    return movies.filter(function(movie){
+        return (movie.director == "Steven Spielberg" && movie.genre.indexOf("Drama") !== -1);
+    }).length;
+}
```

Comment:
Very good solution, again, everything is the same return, one suggestion would be using .includes instead of .indexOf.